JAX build fails when using clang>=19

### Description

As in the title:
```
ERROR: /home/pearu/.cache/bazel/_bazel_pearu/9427f1bc27e04c0336810be16ba11375/external/upb/BUILD:57:11: Compiling upb/upb.c failed: (Exit 1): clang-19 failed: error executing CppCompile command (from target @@upb//:upb) 
  (cd /home/pearu/.cache/bazel/_bazel_pearu/9427f1bc27e04c0336810be16ba11375/execroot/__main__ && \
  exec env - \
    CLANG_COMPILER_PATH=/mnt/md1/pearu/miniconda3/envs/jax-dev-wheels/bin/clang-19 \
    LD_LIBRARY_PATH=/extras/CUPTI/lib64:/home/pearu/miniconda3/envs/jax-dev-wheels/lib: \
    PATH=/home/pearu/.cache/bazelisk/downloads/sha256/c97f02133adce63f0c28678ac1f21d65fa8255c80429b588aeeba8a1fac6202b/bin:/home/pearu/.nix-profile/bin:/home/pearu/miniconda3/envs/jax-dev-wheels/bin:/home/pearu/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/pearu/git/junegunn/fzf/bin \
    PWD=/proc/self/cwd \
  /mnt/md1/pearu/miniconda3/envs/jax-dev-wheels/bin/clang-19 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/k8-opt/bin/external/upb/_objs/upb/upb.pic.d '-frandom-seed=bazel-out/k8-opt/bin/external/upb/_objs/upb/upb.pic.o' -fPIC -iquote external/upb -iquote bazel-out/k8-opt/bin/external/upb '-fvisibility=hidden' -Wno-sign-compare -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-array-parameter '-DMLIR_PYTHON_PACKAGE_PREFIX=jaxlib.mlir.' -Wno-gnu-offsetof-extensions -Qunused-arguments '-Werror=mismatched-tags' -mavx -Wno-gnu-offsetof-extensions -Qunused-arguments '-Werror=mismatched-tags' -mavx -Werror -Wno-long-long -pedantic -Wstrict-prototypes -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/upb/upb/upb.c -o bazel-out/k8-opt/bin/external/upb/_objs/upb/upb.pic.o)
# Configuration: 93788ca9f87097679fa810df961263f6f40b1a0fa0ac1251f1fec4f141e55a1c
# Execution platform: @@local_execution_config_platform//:platform
external/upb/upb/upb.c:192:10: error: defining a type within 'offsetof' is a C23 extension [-Werror,-Wc23-extensions]
  192 |   n &= ~(upb_alignof(upb_arena) - 1);
      |          ^~~~~~~~~~~~~~~~~~~~~~
external/upb/upb/upb.c:183:37: note: expanded from macro 'upb_alignof'
  183 | #define upb_alignof(type) offsetof (struct { char c; type member; }, member)
      |                                     ^~~~~~
/mnt/md1/pearu/miniconda3/envs/jax-dev-wheels/lib/clang/19/include/__stddef_offsetof.h:16:43: note: expanded from macro 'offsetof'
   16 | #define offsetof(t, d) __builtin_offsetof(t, d)
      |                                           ^
1 error generated.
Target //jaxlib/tools:build_wheel failed to build
```

Switching to clang 18 or using clang 19 with `-std=c23`, both lead to a successful build.

### System info (python version, jaxlib version, accelerator, etc.)

<details>

```
>>> import jax
>>> jax.print_environment_info()
WARNING:2025-03-12 16:39:14,406:jax._src.xla_bridge:967: An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jax:    0.5.3.dev20250312+74b4d868e
jaxlib: 0.5.3.dev20250312+74b4d868e
numpy:  2.2.3
python: 3.11.0 | packaged by conda-forge | (main, Jan 14 2023, 12:27:40) [GCC 11.3.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='ex', release='5.4.0-171-generic', version='#189-Ubuntu SMP Fri Jan 5 14:23:02 UTC 2024', machine='x86_64')


$ nvidia-smi
Wed Mar 12 16:39:16 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 560.35.03              Driver Version: 560.35.03      CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 2060 ...    Off |   00000000:17:00.0 Off |                  N/A |
| 17%   45C    P0             35W /  175W |       1MiB /   8192MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA GeForce RTX 2060 ...    Off |   00000000:65:00.0 Off |                  N/A |
| 17%   52C    P0             38W /  175W |       1MiB /   8192MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

</details>

--- a/.bazelrc
+++ b/.bazelrc
@@ -132,6 +132,8 @@ build:clang --copt=-Wno-gnu-offsetof-extensions
 build:clang --copt=-Qunused-arguments
 # Error on struct/class mismatches, since this causes link failures on Windows.
 build:clang --copt=-Werror=mismatched-tags
+# Required when building with clang>=19, see jax-ml/jax#27091
+build:clang --copt=-Wno-error=c23-extensions
 
 # Configs for CUDA
 build:cuda --repo_env TF_NEED_CUDA=1

